README.md

```diff
@@ -5,757 +5,875 @@
-      return this.state;
-    }
-  }
-  goSomewhere(newLocation) {
-    return this.location.set(newLocation);
+  name = String;
+  age = Number;
+
+  changeName(name) {
+    return this.name.set(name);
   }
 }
 
-create(Person, { home: 'Toronto', location: 'San Francisco' }).goHome().state;
-// => { home: 'Toronto', location: 'Toronto', isAtHome: true }
+let homer = create(Person, { name: 'Homer', age: 39 });
 
-create(Person, { home: 'Toronto', location: 'San Francisco' }).goSomewhere('Bangalore').state;
-// => { home: 'Toronto', location: 'Bangalore', isAtHome: false }
+let lisa = homer.changeName('Lisa');
 ```
 
-## Batch Transitions
+## chaining transitions
 
-Transitions can be used to perform multiple transformations in sequence. This is useful when
-you have a deeply nested microstate and you're applying several transformations to one branch of the
-microstate. All of the operations we'll perform before the transformation is complete.
+Transitions can be composed out of any number of subtransitions. This is often referred to as "batch transitions" or "transactions". Let's say that when we authenticate a session, we need to both store the token and indicate that the user is now authenticated. To do this, we can chain transitions. The result of the last operation in the chain will be merged into the microstate.
 
 ```js
-class MyModal {
-  isOpen = Boolean;
-  title = String;
-  content = String;
+class Session {
+  token = String;
+}
 
-  show(title, content) {
+class Authentication {
+  session = Session;
+  isAuthenticated = Boolean;
+
+  authenticate(token) {
     return this
-      .isOpen.set(true),
-      .title.set(title)
-      .content.set(content)
+      .session.token.set(token)
+      .isAuthenticated.set(true);
   }
 }
 
-class MyComponent {
-  modal = MyModal;
-  counter = Number
+class App {
+  authentication = Authentication
 }
 
-create(MyComponent).modal.show('Hello World', 'Rise and shine!').state;
-// => { modal: { isOpen: true, title: 'Hello World', content: 'Rise and shine!' }, counter: 0 }
+let app = create(App, { authentication: {} });
+
+let authenticated = app.authentication.authenticate('SECRET');
+
+authenticated.valueOf();
+//> { authentication: { session: { token: 'SECRET' }, isAuthenticated: true } }
 ```
 
-## Changing structure
+## `initialize` transition
 
-When modeling state machines, it's often helpful to be able to initialize into a particular
-state based on value. This can be accomplished by adding a static `create` method. This method
-should return the next state that you want the node to initialize into.
+The `initialize` transition converts value into a Microstate when a Microstate is being created with the `create` function. The initialize transition is invoked for every microstate that declares one. They are called from top to bottom, meaning that a parent microstate is initalized before the children. This is imporant because the parent microstate can change what children are created.
+
```

Comment:
This jumps into the what of the initialize transition, let's linger just a while on the Why.

> Just as every state machine must begin life in its "start state", so to must every microstate begin life in the right state. For those cases where the start state depends on some logic, there is the `initialize` transition. The `initialize` transition is just like any other transition, except that it will be invoked automatically within `create` on every Microstate that declares one.